src/test/java/com/aerospike/mapper/InterfaceHierarchyTest.java

```diff
@@ -0,0 +1,127 @@
+            this.myName = myName;
+        }
+        @Override
+        public String getName() {
+            return myName;
+        }
+        @Override
+        public String toString() {
+            return "SubClass1: " + myName;
+        }
+    }
+    @AerospikeRecord
+    @NoArgsConstructor
+    @Data
+    public static class SubClass2 implements BaseInterface {
+        @AerospikeKey
+        private String myName;
+        public SubClass2(String myName) {
+            this.myName = myName;
+        }
+        @Override
+        public String getName() {
+            return myName;
+        }
+        @Override
+        public String toString() {
+            return "SubClass2: " + myName;
+        }
+    }
+    
+    @AerospikeRecord(set="container", namespace = "test")
+    @Data
+    public static class Container {
+        @AerospikeKey
+        private long id;
+        private final List<BaseInterface> children;
+        public Container() {
+            this.children = new ArrayList<>();
+            this.id = 1;
+        }
+        public Container(BaseInterface firstChild, BaseInterface ... otherChildren) {
+            this();
+            this.children.add(firstChild);
+            for (int i = 0; i < otherChildren.length; i++) {
+                this.children.add(otherChildren[i]);
+            }
+        }
+    }
+
+    @AerospikeRecord(set="container", namespace = "test")
+    @Data
+    public static class NestedContainer {
+        @AerospikeKey
+        private long id;
+        @AerospikeEmbed
+        private final List<BaseInterface> children;
+        public NestedContainer() {
+            this.children = new ArrayList<>();
+            this.id = 2;
+        }
+        public NestedContainer(BaseInterface firstChild, BaseInterface ... otherChildren) {
+            this();
+            this.children.add(firstChild);
+            for (int i = 0; i < otherChildren.length; i++) {
+                this.children.add(otherChildren[i]);
+            }
+        }
+    }
+
+    @Test
+    public void runTest() {
+        AeroMapper mapper = new AeroMapper.Builder(client).build();
+        Container container = new Container(
+                new SubClass1("Bob"),
+                new SubClass2("Fred"),
+                new SubClass1("Wilma")
+        );
+        
+        mapper.save(container, container.children.get(0), container.children.get(1), container.children.get(2));
+        Container readContainer = mapper.read(Container.class, 1);
+        assertEquals(container, readContainer);
+    }
+
+    @Test
+    public void runNestedTest() {
+        AeroMapper mapper = new AeroMapper.Builder(client).build();
+        NestedContainer container = new NestedContainer(
+                new SubClass1("Bob"),
+                new SubClass2("Fred"),
+                new SubClass1("Wilma")
+        );
+        
+        mapper.save(container, container.children.get(0), container.children.get(1), container.children.get(2));
```

Comment:
Passes using `mapper.save(container);`
What is the motivation behind passing inner objects to the save method?

Reply:
Good catch. I copy-pasted the test with the only difference being the embedding of the children to make sure functionality was what was expected. But you're right, there's no need to explicitly save the children here as they're embedded.